editor/script_editor_debugger.cpp

```diff
@@ -574,7 +574,38 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, const Array &p_da
 		for (int i = 0; i < arr.size(); i++) {
 			p[i] = arr[i];
 			if (i < perf_items.size()) {
-				perf_items[i]->set_text(1, rtos(p[i]));
+
+				float v = p[i];
+				String vs = rtos(v);
+				String tt = vs;
+				switch (Performance::MonitorType((int)perf_items[i]->get_metadata(1))) {
+					case Performance::MONITOR_TYPE_MEMORY: {
+						// for the time being, going above GBs is a bad sign.
+						String unit = "B";
+						if ((int)v > 1073741824) {
+							unit = "G";
+							v /= 1073741824.0;
+						} else if ((int)v > 1048576) {
+							unit = "M";
+							v /= 1048576.0;
+						} else if ((int)v > 1024) {
+							unit = "K";
+							v /= 1024.0;
+						}
+						tt += " bytes";
+						vs = rtos(v) + " " + unit;
```

Comment:
maybe it should limit decimal places.

Reply:
im not sure about that one, since that might also be applied to time measurements for them to look consistent, and currently this does not handle time units since the only ones are from the process and fixed time which are very tiny numbers, im not sure if i should put ms for the and truncate them.